lib/Mojo/Server/Daemon.pm

```diff
@@ -146,7 +147,17 @@ sub _finish {
 sub _listen {
   my ($self, $listen) = @_;
 
-  my $url     = Mojo::URL->new($listen);
+  my $url = Mojo::URL->new($listen);
+
+  croak(
+    "Scheme not specified or not one of http/https in listen location '$listen'"
+  ) if !$url->scheme || $url->scheme !~ /^https*$/;
```

Comment:
```
croak("Scheme need to be http or https in listen location '$listen'") unless $url->scheme =~ /^https?$/;
```
Also note that "s*" == sssssss, while you probably don't want to match "httpssssss".